### ISSUE TITLE
9.1.5: CreateTableEnumerator and ipairs_reverse breaks on first false value in table

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#175]
In 9.1.5.40290 the implementation of `ipairs_reverse` was changed to the following:

```lua
function ipairs_reverse(table)
	local function Enumerator(table, index)
		index = index - 1;
		local value = table[index];
		if value then
			return index, value;
		end
	end
	return Enumerator, table, #table + 1;
end
```

Unfortunately the branch on `if value` won't be entered if the iterator passes over a boolean `false` in the table - leading to no value being returned and the iterator terminating. The same issue impacts the newly added `CreateTableEnumerator` which has a similar check in its iterator as well. This can be verified with the following:

```
/run for i, v in ipairs_reverse({ 1, 2, true, false, 5, 6 }) do print(i, v); end
/run for i, v in CreateTableEnumerator({ 1, 2, false, true, 5, 6 }) do print(i, v); end
```

The expected behaviour is for each `/run` command above to print all 6 values to the chat frame. In the aforementioned 9.1.5 build, only 2 values will be printed for each individual `/run` command.

**Comment by Meorawr** [comment]
Fixed internally, pending a future build.

**Comment by Meorawr** [comment]
Fixed in 9.1.5.40383.
